packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_view.dart

```diff
@@ -13,6 +18,173 @@ class SnapshotView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Text('Content of ${item.name} will be here');
+    return ValueListenableBuilder<bool>(
+      valueListenable: item.isProcessing,
+      builder: (_, isProcessing, __) {
+        if (isProcessing) return const SizedBox.shrink();
+
+        final stats = item.stats;
+        if (stats == null) {
+          return const Center(child: Text('Could not take snapshot.'));
+        }
+
+        return _StatsTable(
+          data: stats,
+          sorting: item.sorting,
+          key: ObjectKey(item),
+        );
+      },
+    );
+  }
+}
+
+class _ClassNameColumn extends ColumnData<HeapStatsRecord> {
+  _ClassNameColumn()
+      : super(
+          'Class',
+          titleTooltip: 'Class name',
+          fixedWidthPx: scaleByFontFactor(100.0),
+          alignment: ColumnAlignment.left,
+        );
+
+  @override
+  String? getValue(HeapStatsRecord record) => record.className;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  String getTooltip(HeapStatsRecord record) => record.fullClassName;
+}
+
+class _InstanceColumn extends ColumnData<HeapStatsRecord> {
+  _InstanceColumn()
+      : super(
+          'Non GC-able\nInstances',
+          titleTooltip: 'Number of instances of the class, '
+              'that have retaining path from the root.',
+          fixedWidthPx: scaleByFontFactor(110.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(HeapStatsRecord record) => record.instanceCount;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+}
+
+class _ShallowSizeColumn extends ColumnData<HeapStatsRecord> {
+  _ShallowSizeColumn()
+      : super(
+          'Shallow\nSize',
+          titleTooltip: 'Total shallow Dart (not native) size of objects.',
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(HeapStatsRecord record) => record.shallowSize;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+
+  @override
+  String getDisplayValue(HeapStatsRecord record) => prettyPrintBytes(
+        getValue(record),
+        includeUnit: true,
+        kbFractionDigits: 1,
+      )!;
+}
+
+class _RetainedSizeColumn extends ColumnData<HeapStatsRecord> {
+  _RetainedSizeColumn()
+      : super(
+          'Retained\nSize',
+          titleTooltip: 'Total size of objects plus objects they retain, '
+              'taking to account only the shortest retaining path for the referenced objects.',
```

Comment:
nit: taking _into_ account

does this size exclude native as well? if so, we should mention that here to match what is done for shallow size